src/test/java/programminglife/parser/GraphParserTest.java

```diff
@@ -70,4 +73,35 @@ public void parseSegmentTest() {
         assertEquals(0, node.getParents().size());
         assertEquals(0, node.getChildren().size());
     }
+
+    @Test
+    public void runTestSuccess() {
+        graphParser.addObserver(this);
+        graphParser.run();
+    }
+
+    @Test
+    public void runTestFailure() {
+        try {
+            faultyGraphParser.addObserver(this);
+            faultyGraphParser.run();
```

Comment:
Also put `fail();` after this to make sure it won't pass when it doesn't throw an exception.